File: src/test/java/dev/retrotv/crypt/twe/lea/LEACBCTest.java
package dev.retrotv.crypt.twe.lea;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import javax.crypto.spec.IvParameterSpec;
import java.security.Key;

import static org.junit.jupiter.api.Assertions.assertEquals;

class LEACBCTest {

    @Test
    @DisplayName("LEACBC-128 암복호화 테스트")
    void leacbc128_test() throws Exception {
        String message = "The lazy dog jumps over the brown fox!";
        LEACBC lea = new LEACBC(128);
        Key key = lea.generateKey();
        IvParameterSpec iv = lea.generateSpec();
        lea.dataPadding();

        byte[] encryptedData = lea.encrypt(message.getBytes(), key, iv);
        String originalMessage = new String(lea.decrypt(encryptedData, key, iv));

        assertEquals(message, originalMessage);
    }

    @Test
    @DisplayName("LEACBC-192 암복호화 테스트")
    void leacbc192_test() throws Exception {
        String message = "The lazy dog jumps over the brown fox!";
        LEACBC lea = new LEACBC(192);
        Key key = lea.generateKey();
        IvParameterSpec iv = lea.generateSpec();
        lea.dataPadding();

        byte[] encryptedData = lea.encrypt(message.getBytes(), key, iv);
        String originalMessage = new String(lea.decrypt(encryptedData, key, iv));

        assertEquals(message, originalMessage);
    }

    @Test
    @DisplayName("LEACBC-256 암복호화 테스트")
    void leacbc256_test() throws Exception {
        String message = "The lazy dog jumps over the brown fox!";
        LEACBC lea = new LEACBC(256);
        Key key = lea.generateKey();
        IvParameterSpec iv = lea.generateSpec();
        lea.dataPadding();

        byte[] encryptedData = lea.encrypt(message.getBytes(), key, iv);
        String originalMessage = new String(lea.decrypt(encryptedData, key, iv));

        assertEquals(message, originalMessage);
    }
}

File: src/test/kotlin/dev/retrotv/crypto/twe/lea/LEACBCTest.kt
package dev.retrotv.crypto.twe.lea

import dev.retrotv.crypto.twe.ParamsWithIV
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.DisplayName
import kotlin.test.Test


internal class LEACBCTest {

    @Test
    @DisplayName("LEACBC-128 암복호화 테스트")
    fun kisa_leacbc128_test() {
        val message = "The lazy dog jumps over the brown fox!"
        val lea = LEACBC(128)
        val key = lea.generateKey()
        val iv = lea.generateSpec()
        val params = ParamsWithIV(key.encoded, iv.iv)

        val encryptedData = lea.encrypt(message.toByteArray(), params)
        val originalMessage = String(lea.decrypt(encryptedData.data, params).data)
        Assertions.assertEquals(message, originalMessage)
    }

    @Test
    @DisplayName("LEACBC-192 암복호화 테스트")
    fun kisa_leacbc192_test() {
        val message = "The lazy dog jumps over the brown fox!"
        val lea = LEACBC(192)
        val key = lea.generateKey()
        val iv = lea.generateSpec()
        val params = ParamsWithIV(key.encoded, iv.iv)

        val encryptedData = lea.encrypt(message.toByteArray(), params)
        val originalMessage = String(lea.decrypt(encryptedData.data, params).data)
        Assertions.assertEquals(message, originalMessage)
    }

    @Test
    @DisplayName("LEACBC-256 암복호화 테스트")
    fun kisa_leacbc256_test() {
        val message = "The lazy dog jumps over the brown fox!"
        val lea = LEACBC(256)
        val key = lea.generateKey()
        val iv = lea.generateSpec()
        val params = ParamsWithIV(key.encoded, iv.iv)

        val encryptedData = lea.encrypt(message.toByteArray(), params)
        val originalMessage = String(lea.decrypt(encryptedData.data, params).data)
        Assertions.assertEquals(message, originalMessage)
    }
}
